vendor/bat-native-ads/src/bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.cc

```diff
@@ -0,0 +1,130 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_events.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_processor_util.h"
+
+#include "base/check.h"
+#include "bat/ads/internal/base/logging_util.h"
+#include "bat/ads/internal/base/search_engine/search_engine_results_page_util.h"
+#include "bat/ads/internal/base/search_engine/search_engine_util.h"
+#include "bat/ads/internal/features/text_embedding_features.h"
+#include "bat/ads/internal/locale/locale_manager.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_info.h"
+#include "bat/ads/internal/ml/pipeline/text_processing/embedding_processing.h"
+#include "bat/ads/internal/resources/contextual/text_embedding/text_embedding_resource.h"
+#include "bat/ads/internal/resources/language_components.h"
+#include "bat/ads/internal/resources/resource_manager.h"
+#include "bat/ads/internal/tabs/tab_manager.h"
+#include "url/gurl.h"
+
+namespace ads::processor {
+
+TextEmbedding::TextEmbedding(resource::TextEmbedding* resource)
+    : resource_(resource) {
+  DCHECK(resource_);
+
+  LocaleManager::GetInstance()->AddObserver(this);
+  ResourceManager::GetInstance()->AddObserver(this);
+  TabManager::GetInstance()->AddObserver(this);
+}
+
+TextEmbedding::~TextEmbedding() {
+  LocaleManager::GetInstance()->RemoveObserver(this);
+  ResourceManager::GetInstance()->RemoveObserver(this);
+  TabManager::GetInstance()->RemoveObserver(this);
+}
+
+bool TextEmbedding::IsEnabled() {
+  return targeting::features::IsTextEmbeddingEnabled();
+}
+
+void TextEmbedding::Process(const std::string& text) {
+  if (!resource_->IsInitialized()) {
+    BLOG(1, "Failed to process text embeddings as resource not initialized");
+    return;
+  }
+
+  const std::string sanitized_text = SanitizeHtml(text);
+  if (sanitized_text.empty()) {
+    BLOG(1, "No text available for embedding");
+    return;
+  }
+
+  ml::pipeline::EmbeddingProcessing* embedding_proc_pipeline = resource_->Get();
+  ml::pipeline::TextEmbeddingInfo text_embedding =
+      embedding_proc_pipeline->EmbedText(sanitized_text);
+  if (text_embedding.embedding.GetNonZeroElementCount() == 0) {
+    BLOG(1, "Failed to embed text");
+    return;
+  }
+
+  const std::string embedding_as_string =
+      text_embedding.embedding.GetVectorAsString();
+  BLOG(9, "Embedding: " << embedding_as_string);
```

Comment:
is this log needed? And if so, do you have example of its output please, thanks

Reply:
I would prefer to leave this one in, since it's the only place to view the embedding without going into the database. Since it's blog-level 9, it shouldn't blow up the logs for most people. Example is below:

Reply:
`BLOG` is only intended to help with diagnosing issues for users

Reply:
BLOG removed